udata/assets.py

```diff
@@ -73,10 +73,15 @@ def from_manifest(app, filename, raw=False, **kwargs):
     path = _manifests[app][filename]
 
     if not raw and cfg.get('CDN_DOMAIN') and not cfg.get('CDN_DEBUG'):
-        prefix = 'https://' if cfg.get('CDN_HTTPS') else '//'
-        if not path.startswith('/'):
+        scheme = 'https' if cfg.get('CDN_HTTPS') else request.scheme
+        prefix = '{}://'.format(scheme)
+        if not path.startswith('/'):  # CDN_DOMAIN has no trailing slash
             path = '/' + path
         return ''.join((prefix, cfg['CDN_DOMAIN'], path))
+    elif not raw and kwargs.get('external', False):
+        if path.startswith('/'):  # request.host_url has a trailing slash
```

Comment:
Same here.

Reply:
Same here:
the `startswith` is on `path` which is appended to `request.host_url`